### DISCUSSION TITLE
Multilingual search implementation

### DISCUSSION THREAD

**Discussion by sarcadass** [algolia/instantsearch#4709] [Q&A] [answered]
Hello,
I'm trying to implement a multilingual search that will only search on certain properties. The different translations are in the same index as you can see in this preview of the Data model:
![AAAA](https://user-images.githubusercontent.com/2452603/113131163-70f35900-924f-11eb-8e12-a3942470d3d8.png)

Depending on the active local of the site I would like to search on the `translations[local]` & `content[local]` keys if they exists.
So if I'm in the `en` local, I would like to search only on `translations['en']` & `content['en']`.

I've tried multiple things following [this guide](https://www.algolia.com/doc/guides/managing-results/optimize-search-results/handling-natural-languages-nlp/how-to/multilingual-search/#option-2-each-record-contains-all-languages) (but it seems to work only if the translations keys are at the root level of the object, which is not the case here because they are contained inside `content` & `translations` objects) and [this guide](https://www.algolia.com/doc/api-reference/api-parameters/queryLanguages/?client=javascript) without success.

Here is a sample of the code with **instantsearch**:
```js
const local = 'en'

// Not working
helper
  .setQuery(helper.state.query)
  .setSettings({ queryLanguages: [local] })
  // or
  .setQueryParameter("queryLanguages", [local])
  .search()

// Also not working
helper
  .setQuery(helper.state.query)
  .setQueryParameter("searchableAttributes", [`translations.${local}`, `content.${local}`])
  // or
  .setQueryParameter("restrictSearchableAttributes", [`translations.${local}`, `content.${local}`])
  .search()
```

What would be the proper way of implementing a Multilingual search in this context ?

**Comment by Haroenv** [comment, marked as answer]
searchable attributes is a setting, not a query parameter, which is why that second doesn't work. Similarly in the first example, you're setting settings, which is done once per index, not per session, and not at the frontend. 

If you'd actually want to search differently, the best way to go is to create an index per language, that otherwise follows the same schema, so you can use the same frontend, just different indices.

Otherwise restrictSearchableAttributes should indeed work, but it might not in the example you sent, because of passing searchableAttributes as a parameter instead of a setting.

Hope that already helps, but more detailed help will be best given by contacting support@algolia.com, since this is irrelevant to InstantSearch itself.

Have a nice day!

**Reply by sarcadass** [reply]
How may I set a **setting** instead of a search **parameter**, I tried to find the answer on [the doc](https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/#widget-param-searchclient) without success.

**Reply by Haroenv** [reply]
you can set a setting using `algoliasearch('appId', 'writeApiKey').initIndex('myIndex').setSettings({...})`, see the doc here: https://www.algolia.com/doc/api-reference/api-methods/set-settings/?client=javascript

The settings can also be set via the dashboard, it doesn't have to be done by an api client per se

**Reply by sarcadass** [reply]
Ok, thank you for the answer, I'll try that later. If I set the settings via the dashboard, I think I could not adapt the settings based on the client local.

**Reply by Haroenv** [reply]
you basically set the settings for all possible searchable attributes, and then you can still use query languages & restrict searchable attributes are search parameters though

**Reply by sarcadass** [reply]
Ok thank you for detailing that point.

**Reply by sarcadass** [reply]
It seems the project I'm working on does not have a sufficient plan to use the restriction search.
I will see to implement an index per language to have a proper result count and lighter indices.
![Frame 1](https://user-images.githubusercontent.com/2452603/113547369-5559cb00-9620-11eb-8b26-fa41599f801e.png)
